libs/geospatial/src/main/java/org/codice/ddf/libs/location/CountryCodeConverter.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.libs.location;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.security.PropertiesLoader;
+
+/**
+ * Utility to convert country codes into various formats.
+ */
+public class CountryCodeConverter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CountryCodeConverter.class);
+
+    private static final String FIPS_TO_ISO3 = "FIPS10-4toISO3.properties";
+
+    public static final String ENGLISH_LANG = "en";
+
+    private static Map<String, String> countryCodesMap;
+
+    static {
+        countryCodesMap = PropertiesLoader.toMap(PropertiesLoader.loadProperties(FIPS_TO_ISO3));
+    }
+
+    /**
+     * Converts FIPS 10-4 into ISO 3166-1 alpha 3.
+     *
+     * @param fipsCountryCode a FIPS 10-4 country code
+     * @return an ISO 3166-1 alpha-3 country code or null if there isn't a valid conversion.
+     */
+    public static String convertFipsToIso3(String fipsCountryCode) {
+        String iso3CountryCode = countryCodesMap.get(fipsCountryCode);
+        if (StringUtils.isBlank(iso3CountryCode)) {
+            LOGGER.debug(
+                    "Could not find a corresponding ISO 3166-1 alpha 3 country code given the FIPS"
+                            + " 10-4 country code: {}",
+                    fipsCountryCode);
+        }
+        return iso3CountryCode;
+    }
+
+    /**
+     * Converts ISO 3166-1 alpha 3 into FIPS 10-4.
+     * <p>
+     * Special case:
+     * ISO 3166-1 alpha 3 country: "PSE" corresponds to FIPS 10-4 countries: "WE" and "GZ",
+     * in this case, the returned list will contain those two countries. In all other cases,
+     * the returned list will contain one country.
+     *
+     * @param iso3alphaCountryCode a FIPS 10-4 country code
+     * @return an ISO 3166-1 country code in a List<String> or an empty list if there isn't a valid
+     * conversion.
+     */
+    public static List<String> convertIso3ToFips(String iso3alphaCountryCode) {
+        if ("PSE".equals(iso3alphaCountryCode)) {
```

Comment:
Since it is possible to have a single alpha 3 code map to multiple FIPS 10-4 codes, maybe we should build another map (See Multimap: http://google.github.io/guava/releases/19.0/api/docs/com/google/common/collect/Multimap.html) from the properties file and not have a special case here. Hard-coded special cases aren't great as it is really easy to not know about them when updating the country code properties file and forget to add and/or remove the hard-coded cases.